modules/auxiliary/scanner/lotus/lotus_domino_hashes.rb

```diff
@@ -157,9 +157,14 @@ def dump_hashes(view_id, cookie, uri)
       }, 25)
 
       if res && res.body
-        short_name = res.body.scan(/<INPUT NAME=\"ShortName\" TYPE=(?:.*) VALUE=\"([^\s]+)"/i).join
-        user_mail = res.body.scan(/<INPUT NAME=\"InternetAddress\" TYPE=(?:.*) VALUE=\"([^\s]+)"/i).join
-        pass_hash = res.body.scan(/<INPUT NAME=\"\$?dspHTTPPassword\" TYPE=(?:.*) VALUE=\"([^\s]+)"/i).join
+        short_name = res.body.scan(/<INPUT NAME=\"ShortName\" TYPE=(?:.*) VALUE=\"([^"]+)\"/i).join
+        user_mail = res.body.scan(/<INPUT NAME=\"InternetAddress\" TYPE=(?:.*) VALUE=\"([^"]+)\"/i).join
+        pass_hash_candidates = res.body.scan(/<INPUT NAME=\"\$?dspHTTPPassword\" TYPE=(?:.*) VALUE=\"([^"]+)\"/i)
```

Comment:
Maybe it would be safer to use a proper parsing engine such as Xpath instead of raw regex?
`Rex::Proto::Http::Response` objects offers a `get_html_document` method that returns a `Nokogiri::XML::Document`. For example (not tested):
```ruby
doc = res.get_html_document
short_name =  doc.xpath('//input[@name="ShortName"]/@value').text
```

Reply:
Yes, that would be safer. I tested it and it seems to work.